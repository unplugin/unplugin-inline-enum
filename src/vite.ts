/**
 * This entry file is for Vite plugin.
 *
 * @module
 */

import unplugin from './index'

/**
 * Vite plugin
 *
 * @example
 * ```ts
 * // vite.config.ts
 * import InlineEnum from 'unplugin-inline-enum/vite'
 *
 * export default defineConfig({
 *   plugins: [InlineEnum()],
 * })
 * ```
 */
const vite = unplugin.vite
export default vite
export { vite as 'module.exports' }
