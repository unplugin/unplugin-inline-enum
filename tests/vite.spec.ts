import path from 'node:path'
import { build } from 'vite'
import { expect, test } from 'vitest'
import UnpluginInlineEnum from '../src/vite'
import type { RollupOutput } from 'rollup'

test('vite', async () => {
  const root = path.resolve(__dirname, 'fixtures')
  const { output } = (await build({
    root,
    build: {
      minify: false,
      rollupOptions: {
        input: [path.resolve(root, 'main.ts')],
      },
      write: false,
    },
    logLevel: 'silent',
    plugins: [
      UnpluginInlineEnum({
        scanDir: path.resolve(__dirname, 'fixtures'),
        scanMode: 'fs',
      }),
    ],
  })) as RollupOutput
  expect(output[0].code).toMatchSnapshot()
})
