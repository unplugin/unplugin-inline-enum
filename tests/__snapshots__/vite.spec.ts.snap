// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite 1`] = `
"const values = ["foo", 100, 4];
console.log("foo", 100, 4);
console.log(values);
"
`;
